src/ape_test/contextmanagers.py

```diff
@@ -0,0 +1,29 @@
+from typing import Optional, Type
+
+from ape.exceptions import ContractLogicError
+
+
+class RevertsContextManager:
+    def __init__(self, expected_message: Optional[str] = None):
+        self.expected_message = expected_message
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type: Type, exc_value: Exception, traceback) -> bool:
+        if exc_type is None:
+            raise AssertionError("Transaction did not revert.")
+
+        if not isinstance(exc_value, ContractLogicError):
+            raise AssertionError(
+                f"Transaction did not revert.\nHowever, exception occurred: {exc_value}"
+            ) from exc_value
+
+        actual = exc_value.revert_message
+
+        if self.expected_message is not None and self.expected_message != actual:
+            raise AssertionError(
+                f"'{self.expected_message}' not found in revert message '{actual}'."
+            )
+
+        return True  # Assertion passes
```

Comment:
wait, why is this returning True? is this just how it integrates with pytest?

Reply:
I copied it from brownie, but you are right, it does not seem to be needed

Reply:
I just found out why it is needed: https://docs.python.org/3/library/contextlib.html

^ From line:  `By returning true values, these callbacks can suppress exceptions the same way context manager __exit__() methods can.`

That implies that returning `True` from `__exit__` causes the exceptions to not get raised, which is what we want here

Reply:
lmao tf

Reply:
Problem something to keep in mind when working with context managers.

Also note that `pytest.raises()` does this, as well as `brownie.reverts()`